ldms/python/ldmsd/parser_util.py

```diff
@@ -0,0 +1,1005 @@
+import os, sys
+import errno
+import json
+import subprocess
+import socket
+import time
+import itertools as it
+import collections
+import ldmsd.hostlist as hostlist
+
+AUTH_ATTRS = [
+    'auth',
+    'conf'
+]
+
+CORE_ATTRS = [
+    'daemons',
+    'aggregators',
+    'samplers',
+    'plugins',
+    'stores'
+]
+
+DEFAULT_ATTR_VAL = {
+    'maestro_comm' : False,
+    'xprt'         : 'sock',
+    'interval'     : 1000000,
+    'auth'         : 'none',
+    'mode'         : 'static'
+}
+
+INT_ATTRS = [
+    'interval',
+    'offset',
+    'reconnect',
+    'flush'
+]
+
+unit_strs = {
+    'ms' : 1000,
+    'us' : 1,
+    'm' : 60000000,
+    's' : 1000000,
+    'h' : 3600000000,
+    'd' : 86400000000
+}
+LDMS_YAML_ERR = 'Error parsing ldms_config yaml file'
+LIST_ERR = 'spec must be a list of dictionaries, specified with "-" in the ldms_config yaml file'
+DICT_ERR = 'spec must be a dictionary, with keys referencing a plugin instance name'
+
+def check_intrvl_str(interval_s):
+    """Check the format of the interval string is correct
+
+    A time-interval string is an integer or float followed by a
+    unit-string. A unit-string is any of the following:
+
+    'us' - microseconds
+    'ms' - milliseconds
+    's'  - seconds
+    'm'  - minutes
+    'h'  - hours
+    'd'  - days
+
+    Unit strings are not case-sensitive.
+
+    Examples:
+    '1.5s' - 1.5 seconds
+    '1.5S' - 1.5 seconds
+    '2s'   - 2 seconds
+    """
+    error_str = f"{interval_s} is not a valid time-interval string\n"\
+                f"'Only a single unit-string is allowed. e.g. '50s40us' is not a valid entry."\
+                f"Examples of acceptable format:\n"\
+                f"'1.5s' - 1.5 seconds\n"\
+                f"'1.5S' - 1.5 seconds\n"\
+                f"'2us'  - 2 microseconds\n"\
+                f"'3m'   - 3 minutes\n"\
+                f"'1h'   - 1 hour\n"\
+                f"'1d'   - 1 day\n"\
+                f"\n"
+    if type(interval_s) == int or type(interval_s) == float:
+        return interval_s
+    if type(interval_s) != str:
+        print(f"{error_str}")
```

Comment:
Error message need to go to stderr, not stdout. I'm not make this comment every where it happens in the code, but it needs to be fixed everywhere that it happens.

Reply:
Addressed in 33deed8